doc/library-reference/inet/mqtt_client.rst

```diff
@@ -10,12 +10,66 @@ MQTT is a publish-subscribe-based lightweight messaging protocol.
           transport layer (normally TCP). That has to be set up using
           channels.
 
+The driver works by running the processing code in a thread which
+communicate with the MQTT broker on one side using channels, and the
+application on the other side using queues.
+
+This means the application has to set up appropriate channels, which
+is already ready to communicate with the MQTT server, e.g. using TCP,
+and the thread running the MQTT client.
```

Comment:
I cannot remember why I designed the MQTT client module to use a thread. If the thread is not necessary it should be removed since it saves RAM. Redesigning the MQTT module, if necssecary, it a separate task not related to this pull request, but I want to mention it here anyway.